ontology/obo/obo.go

```diff
@@ -0,0 +1,276 @@
+	Synonyms      []string            // Optional. Can appear multiple times in an entry. Specifies alternative names for term.
+	XRefs         []string            // Optional. Can appear multiple times in an entry. Specifies an external reference about the term. Ex. Wikipedia:Reproduction
+	AltIds        []string            // Optional. Can appear multiple times in an entry. Specify an alternative GO ID that also points to this entry.
+	Relationships []string            // Optional. Can appear multiple times in an entry. Specifies relationships between ontologies other than 'is_a'.
+	Comments      []string            // Optional. Can appear multiple times in an entry. Stores additional comments about an ontology.
+	OtherFields   map[string][]string // Catch all for other field names.
+	Parents       []*Obo              // Parent nodes of this ontology.
+	Children      []*Obo              // Child nodes of this ontology.
+	SubTreeSize   int                 //Total number of descendents.
+}
+
+// Header encodes the header of an Obo file. Header lines are not marked by a leading character,
+// but are defined as lines preceding the first instance of '[Term]'.
+type Header struct {
+	Text []string
+}
+
+// IsADescription contains the information in the "IsA" field of an Obo record.
+type IsADescription struct {
+	ParentId   string
+	ParentInfo []string
+}
+
+// String is a method for IsADescription which formats an input struct as a string.
+func (d IsADescription) String() string {
+	return IsADescriptionToString(d)
+}
+
+// IsADescriptionToString converts an IsADescription struct into a string.
+func IsADescriptionToString(d IsADescription) string {
+	var answer = ""
+	answer = answer + d.ParentId
+	for i := range d.ParentInfo {
+		answer = answer + " " + d.ParentInfo[i]
+	}
+	return answer
+}
+
+// ReadHeader processes the contiguous header from an EasyReader
+// and advances the Reader past the header lines. Note that Header lines are not marked by a leading\
+// character, but are defined as lines preceding the first instance of '[Term]'.
+func ReadHeader(file *fileio.EasyReader) Header {
+	var answer Header
+	var done bool
+	var line string
+	var err error
+	var peek []byte
+
+	// we will add text to the header as long as we do not encounter the string "[Term]", which signifies a new entry.
+	for peek, err = file.Peek(6); err == nil && !peekIsTermLine(peek) && !done; peek, err = file.Peek(6) {
+		line, done = fileio.EasyNextLine(file)
+		answer.Text = append(answer.Text, line)
+	}
+
+	return answer
+}
+
+// peekIsTermLine returns true if an input slice of bytes is equal to "[Term]", which is the signifier of a
+// new ontology element.
+func peekIsTermLine(peek []byte) bool {
+	if len(peek) != 6 {
+		log.Fatalf("Error: peekIsTermLine expected 6 bytes. Received %v.\n", len(peek))
+	}
+	peekString := string(peek)
+	return peekString == "[Term]"
+}
+
+// Read creates a slice of Obo structs and a Header struct from an input filename.
```

Comment:
Do we want a list without the tree.  Maybe the read function should build the tree and return the tree?